snowpack/src/util.ts

```diff
@@ -25,6 +25,9 @@ export const LOCKFILE_NAME = 'snowpack.deps.json';
 // We need to use eval here to prevent Rollup from detecting this use of `require()`
 export const NATIVE_REQUIRE = eval('require');
 
+// We need to use an external file here to prevent Typescript/Rollup from modifying `require` and `import`
+export const REQUIRE_OR_IMPORT = require('../assets/require-or-import.js');
```

Comment:
Open to pushback here, but would [Rollup’s replace plugin](https://github.com/rollup/plugins/tree/master/packages/replace#readme) work better? I know we already have `eval('require')` above (which maybe could be replaced in the same way), but I’m nervous about writing code _for_ our tooling like this.

Reply:
I went down that route first. Rollup executes plugins before transforming to CJS, so the `import` statement was still getting converted to `require`. Looked for Rollup issue reports as well, but couldn't find much. The `postbundle` script is where I landed but I'm open to other ideas.

Reply:
No, that’s fine. I’ve only used it for valid-ish stuff, but had to get around TS/other tooling; haven’t used it for reserved words like `import()`. That’s annoying that it can’t run at the very end. I’m fine with keeping as the script you have!

Reply:
I definitely agree that we're fighting the tooling here. 😞 
I hope we can revisit this soon when the tooling catches up to Node's native ESM.

Reply:
If this is only meant to survive until Node v10 drops (~2 months) and we can assume 100% ESM, I'm okay with it as well. Maybe add a note in the code to remove once we drop our Node v10 support to remind us?

Reply:
Unfortunately Node v12+ only moves us out of the CJS-only world and into a fully mixed CJS/ESM world... I don't think we can assume all users will be using ESM, just that they will have that option. 

However, it probably makes philosophical sense in the Node v12+ world for Snowpack to encourage (or enforce) projects using `"type": "module"`. We could simplify this then—I'll add a note.

Reply:
Yea I guess that's true, especially for existing plugins. My hope is that we can push config files to ESM sooner than later. Either way, this isn't blocking for me

Reply:
Does Node 10 support dynamic import? I just tried and I get errors.